### ISSUE TITLE
Traceback if you do transformation, line counting and sorting

### ISSUE THREAD

**Issue by McRussian** [ClickHouse/ClickHouse#43030]
There is such a table in the database
"CREATE TABLE IF NOT EXISTS auth.sessions
    (
..
        start_ts UInt32,
..
    )
    PARTITION BY toYYYYMMDD(toDate(start_ts))
    PRIMARY KEY start_ts
    ORDER BY (start_ts)
    TTL toDate(start_ts) + toIntervalDay(180)
"
if you make a request that has a transformation, function COUNT and  sorting, then a traceback occurs

SELECT
    COUNT(*) OVER () AS rows_count,
    toFloat64(toYYYYMMDDhhmmss(toDateTime(start_ts))) AS qwe
FROM auth.sessions
ORDER BY qwe ASC
LIMIT 1

Query id: dc8a408d-b894-4fba-99b8-3f469e6765fc

0 rows in set. Elapsed: 0.003 sec. 

Received exception from server (version 22.3.2):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column start_ts in block. There are only columns: toFloat64(toYYYYMMDDhhmmss(toDateTime(start_ts))), count() OVER (). (NOT_FOUND_COLUMN_IN_BLOCK)

If you do not sort or call a function COUNT, then everything works fine

ClickHouse client version 22.3.2.1.

**Comment by socketpair** [comment]
```sql
CREATE TABLE qwe
(
    `start_ts` UInt32
)
ENGINE = MergeTree
ORDER BY start_ts
```

This works OK:
```sql
SELECT
    COUNT(*) OVER () AS rows_count,
    start_ts AS qqq
FROM qwe
ORDER BY qqq ASC
```

This one does not:
```sql
SELECT
    COUNT(*) OVER () AS rows_count,
    toDateTime(start_ts) AS qqq
FROM qwe
ORDER BY qqq ASC
```

**Comment by socketpair** [comment]
This one also WORKS
```sql
SELECT
    COUNT(*) OVER () AS rows_count,
    toDateTime(start_ts) AS qqq
FROM qwe
```

**Comment by den-crane** [comment]
it was fixed in 22.9 https://github.com/ClickHouse/ClickHouse/pull/39354
https://fiddle.clickhouse.com/b2ed780a-d535-4ea9-800c-189ecfccf0f7
